clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/WebSocketTransport.java

```diff
@@ -45,7 +47,12 @@ private String formatUrl(String url) {
         logger.log(LogLevel.Debug, "Starting Websocket connection.");
         this.webSocketClient = client.createWebSocket(this.url, this.headers);
         this.webSocketClient.setOnReceive((message) -> onReceive(message));
-        this.webSocketClient.setOnClose((code, reason) -> onClose(code, reason));
+        this.webSocketClient.setOnClose((code, reason) -> {
+            if (onClose != null) {
```

Comment:
This shouldn't be possible

Reply:
For testing the transport without a HubConnection